tests/test_requirements.py

```diff
@@ -140,6 +141,128 @@ async def test_get_integration_with_requirements(hass):
     ]
 
 
+async def test_get_integration_with_missing_dependencies(hass):
+    """Check getting an integration with missing dependencies."""
+    hass.config.skip_pip = False
+    mock_integration(
+        hass,
+        MockModule("test_component_after_dep"),
+    )
+    mock_integration(
+        hass,
+        MockModule(
+            "test_component",
+            dependencies=["test_component_dep"],
+            partial_manifest={"after_dependencies": ["test_component_after_dep"]},
+        ),
+    )
+    with pytest.raises(loader.IntegrationNotFound):
+        await async_get_integration_with_requirements(hass, "test_component")
+
+
+async def test_get_integration_with_missing_after_dependencies(hass):
+    """Check getting an integration with missing after_dependencies."""
+    hass.config.skip_pip = False
+    mock_integration(
+        hass,
+        MockModule(
+            "test_component",
+            partial_manifest={"after_dependencies": ["test_component_after_dep"]},
+        ),
+    )
+    integration = await async_get_integration_with_requirements(hass, "test_component")
+    assert integration
+    assert integration.domain == "test_component"
+
+
+async def test_get_integration_with_missing_dependencies_first(hass):
+    """Check getting an integration with missing dependencies raising an exception first."""
+    hass.config.skip_pip = False
+    mock_integration(
+        hass,
+        MockModule(
+            "test_component",
+            dependencies=["test_component_dep"],
+            partial_manifest={"after_dependencies": ["test_component_after_dep"]},
+        ),
+    )
+    mock_gather = Future()
+    mock_gather.set_result(
+        [
+            loader.IntegrationNotFound("test_component_dep"),
+            mock_integration(
+                hass,
+                MockModule("test_component_after_dep"),
+            ),
+        ]
+    )
+    with patch(
+        "homeassistant.requirements.asyncio.gather",
```

Comment:
Generally we don't allow mocking `asyncio.*`. Can you mock `async_get_integration` instead?

Reply:
I don't think so.  To test this we need to have a specific order to the return from gather so the exception comes from after_dependencies. Do you know how to do a variable patch so the return changes based on integration name?

Reply:
Oh the variable patch won't work because of the order requirement.

Reply:
https://docs.python.org/3/library/unittest.mock-examples.html#side-effect-functions-and-iterables

Reply:
I removed the need for the tests.